### DISCUSSION TITLE
Object storage

### DISCUSSION THREAD

**Discussion by rupurt** [apache/iggy#466] [Ideas]
Howdy,

Awesome looking project. Are there any plans to enable object storage for the persistence layer?

**Comment by spetz** [comment]
Hey, thanks, glad you like it!
Speaking of the object storage, we'd like to implement such feature for the data archivization, as described [here](https://github.com/iggy-rs/iggy/issues/290).

**Reply by kxxu** [reply]
kafka's hierarchical storage is not a good design. During fault recovery, a large number of data copies are generated, affecting cluster stability. Using object storage as shared storage reduces storage costs and eliminates complex data replication in cluster mode
